docker/setup.sh

```diff
@@ -3,13 +3,15 @@
 PASSWORD=${PASSWORD:-1234567890}
 
 wasmd init --chain-id=testing testing
+# staking/governance token is hardcoded in config, change this
+sed -i 's/"stake"/"ustake"/' "$HOME"/.wasmd/config/genesis.json
```

Comment:
no similar diff for cosm -> ucosm? Is the token created by the first account receiving it?

Reply:
The account balancea are defines un the creaste account command below.

The default genesis hard codes stake as token for staking validators and voting in governance. Sed fixes that (no cli flag)

Reply:
Actually, maybe we make both denoms args to this script, so we control in the package where we run in CI, not here. And then default to these denoms if $1 and $2 are undefined

Reply:
But where is the COSM token configuted? At which point does the token come into existence?

Reply:
When you add it to a genesis account.

There is no explicit definition of tokens or minting.
They look at balances in genesis and those are the tokens on the chain.

I know, you miss weave :)

Reply:
Okay cool, thanks, that answers my question.

And it answers much more than that. In order to get a list of chain tokens, you need to iterate over all accounts on the chain 🤯

Reply:
They added a supply module recently that basically does that iteration. I should give a list of denoms, but no fractional digits or other info